server/src/main/java/org/apache/iotdb/db/mpp/sql/statement/sys/AuthorStatement.java

```diff
@@ -0,0 +1,263 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.mpp.sql.statement.sys;
+
+import org.apache.iotdb.db.metadata.path.PartialPath;
+import org.apache.iotdb.db.mpp.sql.constant.StatementType;
+import org.apache.iotdb.db.mpp.sql.statement.Statement;
+import org.apache.iotdb.db.mpp.sql.statement.StatementVisitor;
+
+public class AuthorStatement extends Statement {
+
+  private final AuthorStatement.AuthorType authorType;
+  private String userName;
+  private String roleName;
+  private String password;
+  private String newPassword;
+  private String[] privilegeList;
+  private PartialPath nodeName;
+
+  /**
+   * AuthorOperator Constructor with AuthorType.
+   *
+   * @param type author type
+   */
+  public AuthorStatement(AuthorStatement.AuthorType type) {
+    super();
+    authorType = type;
+    statementType = StatementType.AUTHOR;
+  }
+
+  /**
+   * AuthorOperator Constructor with OperatorType.
+   *
+   * @param type statement type
+   */
+  public AuthorStatement(StatementType type) {
+    super();
+    authorType = null;
+    statementType = type;
+  }
+
+  public AuthorStatement.AuthorType getAuthorType() {
+    return authorType;
+  }
+
+  public String getUserName() {
+    return userName;
+  }
+
+  public void setUserName(String userName) {
+    this.userName = userName;
+  }
+
+  public String getRoleName() {
+    return roleName;
+  }
+
+  public void setRoleName(String roleName) {
+    this.roleName = roleName;
+  }
+
+  public String getPassWord() {
+    return password;
+  }
+
+  public void setPassWord(String password) {
+    this.password = password;
+  }
+
+  public String getNewPassword() {
+    return newPassword;
+  }
+
+  public void setNewPassword(String newPassword) {
+    this.newPassword = newPassword;
+  }
+
+  public String[] getPrivilegeList() {
+    return privilegeList;
+  }
+
+  public void setPrivilegeList(String[] authorizationList) {
+    this.privilegeList = authorizationList;
+  }
+
+  public PartialPath getNodeName() {
+    return nodeName;
+  }
+
+  public void setNodeNameList(PartialPath nodePath) {
+    this.nodeName = nodePath;
+  }
+
+  public enum AuthorType {
+    CREATE_USER,
+    CREATE_ROLE,
+    DROP_USER,
+    DROP_ROLE,
+    GRANT_ROLE,
+    GRANT_USER,
+    GRANT_ROLE_TO_USER,
+    REVOKE_USER,
+    REVOKE_ROLE,
```

Comment:
why does not use AuthorType  in AuthorOperator?

Reply:
I will make changes